File: IceSync.Api/Domain/Entities/Workflow.cs
using System.ComponentModel.DataAnnotations;

namespace IceSync.Api.Domain.Entities
{
    public class Workflow
    {
        [Key]
        public int WorkflowId { get; set; }
        
        [Required]
        public string WorkflowName { get; set; } = string.Empty;
        
        public bool IsActive { get; set; }
        
        public string MultiExecBehavior { get; set; } = string.Empty;
    }
}

File: IceSync.Web/src/types/Workflow.ts
export interface Workflow {
  workflowId: number;
  workflowName: string;
  isActive: boolean;
  multiExecBehavior: string;
}

export interface SyncResult {
  message: string;
  inserted: number;
  deleted: number;
  updated: number;
}

export interface RunWorkflowResult {
  message: string;
}
